scripts/check_format_cpp.sh

```diff
@@ -41,6 +41,16 @@ done
 # stop right here if clang-format does not exist in $PATH
 command -v $clang_format_executable >/dev/null 2>&1 || { echo >&2 "clang-format executable '$clang_format_executable' not found.  Aborting."; exit 1; }
 
+clang_format_version="$(${clang_format_executable} --version)"
+clang_format_major_version=$(echo "${clang_format_version}" | sed 's/^[^0-9]*\([0-9]*\).*$/\1/g')
+clang_format_minor_version=$(echo "${clang_format_version}" | sed 's/^[^0-9]*[0-9]*\.\([0-9]*\).*$/\1/g')
+
+if [ "${clang_format_major_version}" -ne 14 ] || [ "${clang_format_minor_version}" -ne 0 ]; then
+  echo "***   This indent script requires clang-format version 14.0,"
+  echo "***   but version ${clang_format_major_version}.${clang_format_minor_version} was found instead."
```

Comment:
```suggestion
clang_format_version="$(${clang_format_executable} --version | cut -f 3 -d ' ')"
if [ "${clang_format_major_version}" -ne "14.0.0" ]; then
  echo "***   This indent script requires clang-format version 14.0.0,"
  echo "***   but version ${clang_format_version} was found instead."
```

Reply:
I am seeing
> $ clang-format --version
clang-format version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)

which could not be used with your suggestion,

Reply:
Would still prefer something simpler for the maintenance, as the regular expressions are really complex.

Assuming that all clang versions are of type `a.b.c-<label>`, the following should work:
```suggestion
clang_format_required_version="14.0.0"
clang_format_version="$(${clang_format_executable} --version | awk -F '[ -]' '{print $3}')"
if [ "${clang_format_major_version}" -ne "${clang_format_required_option}" ]; then
  echo "***   This indent script requires clang-format version ${clang_format_required_version},"
  echo "***   but version ${clang_format_version} was found instead."
```